### ISSUE TITLE
Color swatch for multi website view showing wrong on frontend

### ISSUE THREAD

**Issue by ghost** [magento/magento2#32108]
### Preconditions (*)
Magento 2.4.1
PHP 7.3.26-1+ubuntu20.04.1+deb.sury.org+1 (cli) (built: Jan 13 2021 08:01:31) ( NTS )

1. Create 2 websites with 1 store view each 

### Steps to reproduce (*)

1. Create 2 websites with 1 store view each 
2. Create configurable select attribute (eg color) with different store view simple product E.g RED From the First website and YELLOW From another website.
3. Now go to the frontend, default store, and load any category page with configurable products with color as one of the attributes, or directly go to the product page
4. You can see both swatches color **RED and Yellow** still Yellow product is not assigned to a current website. So Yellow must not be visible in Website 1 as it's assigned only to website 2. 

### Expected result (*)
1.  Color Swatch must be displayed according to the assigned website. as per the above steps if a user is on website 1 then a yellow swatch must not be visible because that simple product is not assigned to the current website.

### Actual result (*)

Color Swatches must be displayed according to the simple product assigned website. on the product page or category page all color swatches visible which is not assigned not the current visitor website. so website filter on a configurable child is not working it must work. 
![image](https://user-images.githubusercontent.com/22772614/107625206-fca22d80-6c81-11eb-8504-30626fcda4e6.png)

**Comment by navinbhudiya** [comment]
any update here?

**Comment by AndrewPenry** [comment]
We experienced this issue as well.

We set up 2 websites and a configurable product with size as the configuration. 5 simple products:

T-Shirt (Configurable, Website 1 and Website 2):
S, M, L on Website 1 only (example.com)
XL, XXL on Website 2 only (bigandtall.example.com)

All swatches show on both websites. We found that if we switched the theme to Blank, the swatches would show correctly for a while, but would eventually revert back to showing all swatches after a few orders had been placed.

More disturbingly, if Share Customer Accounts is "Global" then Reorder can put items from the wrong website into the cart. It seems for both the display of the swatches and the reorder function, permissions are checked against the configurable product, not the simple product.
